service/service.py

```diff
@@ -120,6 +120,28 @@ def get_active_related_products(id):
     ]
 
     return make_response(jsonify(result), status.HTTP_200_OK)
+    
+######################################################################
+# QUERY RECOMMENDATIONS BY ID AND TYPE
+######################################################################
+@app.route('/recommendations/<int:id>/type/<int:typeid>', methods=['GET'])
+def get_related_products_with_type(id, typeid):
+    app.logger.info("Query type: %s recommendations for id: %s", typeid, id)
+    recommendations = Recommendation.find_by_id_type(id, typeid)
+
+    if not recommendations:
+        raise NotFound("Type {} recommendations for product {} not found".format(typeid, id))
```

Comment:
This line was not covered in unit tests, could you please add a test case to capture this line

Reply:
You can take a look at line 81 in tests/test_recommendation.py for reference